modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/ext/SwaggerExtension.java

```diff
@@ -1,14 +1,16 @@
 package com.wordnik.swagger.jaxrs.ext;
 
-import com.wordnik.swagger.annotations.ApiOperation;
-import com.wordnik.swagger.models.parameters.Parameter;
-
 import java.lang.annotation.Annotation;
-import java.util.*;
 import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.models.parameters.Parameter;
 
 public interface SwaggerExtension {
   String extractOperationMethod(ApiOperation apiOperation, Method method, Iterator<SwaggerExtension> chain);
-  List<Parameter> extractParameters(Annotation[] annotations, Class<?> cls, boolean isArray, Set<Class<?>> classesToSkip, Iterator<SwaggerExtension> chain);
-  boolean shouldIgnoreClass(Class<?> cls);
-}
\ No newline at end of file
+  List<Parameter> extractParameters(List<Annotation> annotations, Type type, Set<Type> typesToSkip, Iterator<SwaggerExtension> chain);
```

Comment:
Changed `Class<?>` to `Type`, `Set<Class<?>>` to `Set<Type>`.
`isArray` is redundant as it's known from `Type`.